From 23a322c782ba44b07a30d84b437ed5e57088a118 Mon Sep 17 00:00:00 2001
From: Ildar Abubakirov <ildar.abubakirov@componentality.com>
Date: Sat, 30 Mar 2013 10:13:17 +0400
Subject: [PATCH 2/4] ath: allow channels on 5.9GHz band

* new internal ath rule for 5.9GHz band channles with 20Mhz channel
  width. Both ath9k and ath5k drivers affected.

Signed-off-by: Ildar Abubakirov <ildar.abubakirov@componentality.com>
---
 drivers/net/wireless/ath/regd.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/drivers/net/wireless/ath/regd.c b/drivers/net/wireless/ath/regd.c
index 2f7cdff..cd3a9bc 100644
--- a/drivers/net/wireless/ath/regd.c
+++ b/drivers/net/wireless/ath/regd.c
@@ -1,5 +1,6 @@
 /*
  * Copyright (c) 2008-2009 Atheros Communications Inc.
+ * Copyright (c) 2012-2013 Ildar Abubakirov, Componentality Oy
  *
  * Permission to use, copy, modify, and/or distribute this software for any
  * purpose with or without fee is hereby granted, provided that the above
@@ -52,6 +53,10 @@ static int __ath_regd_init(struct ath_regulatory *reg);
 #define ATH9K_5GHZ_5725_5850	REG_RULE(5725-10, 5850+10, 40, 0, 30,\
 				NL80211_RRF_PASSIVE_SCAN | NL80211_RRF_NO_IBSS)
 
+/* 5.85 - 5.95 GHz frequency band */
+#define ATH9K_DSRC_5850_5950	REG_RULE(5850, 5930, 20, 0, 43,\
+/* 172 */		NL80211_RRF_PASSIVE_SCAN)
+
 #define ATH9K_2GHZ_ALL		ATH9K_2GHZ_CH01_11, \
 				ATH9K_2GHZ_CH12_13, \
 				ATH9K_2GHZ_CH14
@@ -93,6 +98,7 @@ static const struct ieee80211_regdomain ath_world_regdom_64 = {
 	REGD_RULES(
 		ATH9K_2GHZ_CH01_11,
 		ATH9K_5GHZ_NO_MIDBAND,
+		ATH9K_DSRC_5850_5950,
 	)
 };
 
-- 
1.7.7.6

